docs/source/dev/backend.rst

```diff
@@ -3,4 +3,269 @@
 How to Write a Backend
 ======================
 
-*TBD*
+Adding support for additional types of file storage or data transportation is possible by creating a backend.
+Backend design has been kept independent of the openPMD-specific logic that maintains all constraints within a file.
+This should allow easy introduction of new file formats with only little knowledge about the rest of the system.
+
+
+File Formats
+------------
+To get started, you should create a new file format in ``include/openPMD/IO/Format.hpp`` representing the new backend.
+Note that this enumeration value will never be seen by users of openPMD-api, but should be kept short and concise to
+improve readability.
+
+.. code-block:: cpp
+
+    enum class Format
+    {
+         JSON
+    };
+
+In order to use the file format through the API, you need to provide unique and characteristic filename extensions that
+are associated with it. This happens in ``src/Series.cpp``:
+
+.. code-block:: cpp
+
+    Format
+    determineFormat(std::string const& filename)
+    {
+        if( auxiliary::ends_with(filename, ".json") )
+            return Format::JSON;
+    }
+
+.. code-block:: cpp
+
+    std::string
+    cleanFilename(std::string const& filename, Format f)
+    {
+        switch( f )
+        {
+            case Format::JSON:
+                return auxiliary::replace_last(filename, ".json", "");
+        }
+    }
+
+Unless your file format imposes additional restrictions to the openPMD constraints, this is all you have to do in the
+frontend section of the API.
+
+IO Handler
+----------
+Now that the user can specify that the new backend is to be used, a concrete mechanism for handling IO interactions is
+required. We call this an ``IOHandler``. It is not concerned with any logic or constraints enforced by openPMD, but
+merely offers a small set of elementary IO operations.
+
+On the very basic level, you will need to derive a class from ``AbstractIOHandler``:
+
```

Comment:
I you like, you can include the interface of `AbstractIOHandler` here as follows:

```rst
.. doxygenclass:: openPMD::AbstractIOHandler
   :project: openPMD
   :members:
   :undoc-members:
```